src/cml.js

```diff
@@ -0,0 +1,103 @@
+const { execSync } = require('child_process');
+const git_url_parse = require('git-url-parse');
+
+const Gitlab = require('./drivers/gitlab');
+const Github = require('./drivers/github');
+const { upload, exec } = require('./utils');
+
+const uri_no_trailing_slash = (uri) => {
+  return uri.endsWith('/') ? uri.substr(0, uri.length - 1) : uri;
```

Comment:
Are there `triml/trimr` or `rstrip/lstrip` in JS?

Reply:
hahaha no! I made one a few iterations before but I removed it. We could use underscore or a library like that